delete_me_proxies_example/HumanReadableCleaningSteps.Rmd

```diff
@@ -0,0 +1,118 @@
+4. **Harmonizing to generic fractions.** Using the `ResultSampleFractionText` field, group fractions into 4 types: "filtered", "unfiltered", "particulate", and "unknown". Replace the values in `ResultSampleFractionText` with these new generic fractions. Use "filtered" when `ResultSampleFractionText` was "Dissolved", "Filtered, lab", or "Filtered, field". Use "unfiltered" when `ResultSampleFractionText` was "Total", "Total Recoverable", "Total Recovrble", or "Recoverable". Use "particulate" when `ResultSampleFractionText` was "Suspended". Use "unknown" when `ResultSampleFractionText` was missing or said "None", "Pot. Dissolved", "Filterable", "Non-filterable", or "Non-Filterable (Particle)". If `ResultSampleFractionText` matched none of these things, it just kept `ResultSampleFractionText` as-is (we can use this as a way to catch scenarios we haven't seen since there should only be 4 unique fractions returned). Then, add a column called `flag_fracAmbiguous` that says TRUE when `ResultSampleFractionText` is "unknown".
+5. **Removing duplicates.** We defined a unique observation by having distinct combinations of the following columns: `OrganizationIdentifier`, `MonitoringLocationIdentifier`, `ActivityStartDate`, `ActivityStartTime.Time`, `CharacteristicName`, `ResultSampleFractionText`. If any observation had an exact match of the values in all 6 of these columns, then it was treated as a duplicate and removed.
+
+<br>
+
+There were additional criteria applied specifically to individual contaminant groups. For both HABS and METALS, we kept only surface water sites (removed an observation if the value in `MonitoringLocationTypeName` was "Well"). While the data was only queried for `sampleMedia` of either "water" or "Water", some of the METALS data had activity media that were still sediment. So, for METALS, we did one additional filter to remove any observations where the value in `ActivityMediaName` or the value in `ResultSampleFractionText` was "Bed Sediment".
+
+# Below is the code that does the numbered cleaning steps above
+
+\footnotesize
+
+```{r eval=FALSE}
+# Some cleaning functions
+
+checkBelowDetect <- function(df, true_output, false_output) {  
```

Comment:
One thing we can discuss for our pipeline is whether we want the functions to find and omit potentially problematic data points, or whether we simply want to perform checks and add data flags so that a user would ultimately decide how to filter the resulting dataset for their purposes.

Meg and Jenny told me that their approach was to perform checks and add flags, but they intentionally left the dataset with the same dimensions as the initial WQP query. I thought that was interesting.

Reply:
I have also seen this approach from people, especially those who don't have as strong of a background in data reproducibility. There is a tendency to preserve because they hesitate to remove data when they may not be able to get it back. I am less conservative about removing bc I feel pretty confident about getting it back, but maybe there is a hybrid where we flag in one step and then later use the flags to actually filter?